### PR TITLE
Fix PHP 7.4 bug

### PR DESCRIPTION
**Pull request by neoteknic**
Fix bug in PHP 7.4

### CODE REVIEW

**Comment by veneliniliev** [comment]
please merge this

**Comment by reytarovskiy** [comment]
Merge please

**Comment by Michaeltlee** [comment]
Addresses https://github.com/dompdf/dompdf/issues/2003

**Comment by animemixedworldgithub** [comment]
Any plans for this to be merged soon? Thanks

**Comment by kosmonowt** [comment]
OK, @bsweeney thanks for this! Meanwhile in another part of the CPDF-Universe we stumbled upon another guy like this.

Line 2598 (the example below starts from 2591) will eventually crash exactly like line 2545.
```php
                            if ($c >= 0) {
                                // Set values in CID to GID map
                                if ($c >= 0 && $c < 0xFFFF && $glyph) {
                                    $cidtogid[$c * 2] = chr($glyph >> 8);
                                    $cidtogid[$c * 2 + 1] = chr($glyph & 0xFF);
                                }

                                if ($c != hexdec($n)) {
                                    $data['codeToName'][$c] = $n;
                                }
                                $data['C'][$c] = $width;
                            } else {
                                $data['C'][$n] = $width;
                            }
```

I propose to change Line 2598 similar to:

`if (ctype_xdigit($c) && ctype_xdigit($n)) {`

Can someone confirm this?

File: lib/Cpdf.php [hunk 1]
```diff
@@ -2542,7 +2542,7 @@ private function openFont($font)
                             $width = floatval($dtmp['WX']);
 
                             if ($c >= 0) {
-                                if ($c != hexdec($n)) {
+                                if (ctype_xdigit($c) && ctype_xdigit($n)) {
```
Comment (hacioguz) [inline lib/Cpdf.php]: fixed... also 2599...

**Comment by rockmanvnx6** [comment]
Modified both 2545 and 2599 still raise a warning:
```
 *
     * @param mixed $content Content that can be cast to string
     *
     * @return $this
     *
     * @throws \UnexpectedValueException
     */
    public function setContent($content)
    {
        if (null !== $content && !\is_string($content) && !is_numeric($content) && !\is_callable([$content, '__toString'])) {
            throw new \UnexpectedValueException(sprintf('The Response content must be a string or object implementing __toString(), "%s" given.', \gettype($content)));
        }
 
        $this->content = (string) $content;
 
        return $this;
    }
 
    /**
     * Gets the current response content.
```

```
"The Response content must be a string or object implementing __toString(), "object" given."
```

Test code:
```
$pdf = PDF::loadHTML("test");
return $pdf->setPaper('a4', 'landscape')->setWarnings(false)->save('invoice.pdf');
```

**Comment by bsweeney** [comment]
Has anyone submitted a PR for these additional changes or should I just go ahead and commit them?

**Comment by bsweeney** [comment]
@rockmanvnx6 this PR wasn't really a general PHP 7.4 fix just a specific correction. Can you start a new issue so we can track that issue separately?

**Comment by bsweeney** [comment]
FYI this PR didn't quite do what was required. The modification replaces the existing logic, but the condition used should wrap the existing logic. I'll push an update along that addresses this specific instance plus other areas where it could be problematic.

**Pull request merged**